src/Orleans/Core/GrainClient.cs

```diff
@@ -381,6 +381,15 @@ public static Streams.IStreamProvider GetStreamProvider(string name)
             return RuntimeClient.Current.CurrentStreamProviderManager.GetProvider(name) as Streams.IStreamProvider;
         }
 
+        public delegate void ConnectionToClusterLostHandler(object sender, EventArgs e);
```

Comment:
Not necessarily. If the recommendation is to reinitialize the client upon a connection loss, then there is no need to a "gained back" event I think.